core/meta/src/TProtoClass.cxx

```diff
@@ -262,7 +279,8 @@ Bool_t TProtoClass::FillTClass(TClass* cl) {
    // We need to fill enums one by one to initialise the internal map which is
    // transient
    {
-      auto temp = new TListOfEnums();
+      auto temp = cl->fEnums.load() ? cl->fEnums.load() :
+                  IsFromRootCling() ? new TListOfEnums() : new TListOfEnumsWithLock();
```

Comment:
Is the lock so expensive to warrant the extra code complexity?

Reply:
Probably not, especially since the mutex are probably not even on in the `IsFromRootCling`.  This one liners is the easiest fix and so we should remove this complexity in a different PR.  See https://github.com/root-project/root/issues/10339 for tracking.